Reorganize release notes for release/10.5.0 (#5103)
woocommerce/woocommerce-gateway-stripe: readme.txt

== Changelog ==

= 10.5.0 - xxxx-xx-xx =

**Important Fixes and Updates**
* Fix - Allow subscription edits when mandates are present, except for mandates from India
* Fix - Prevent fatal error when order ID in webhook references a refund
* Fix - Hide all payment methods on Add Payment Method page when saved payment methods are disabled
* Update - Add missing metadata to checkout session objects when processing webhook events

> * Update - Include additional customer information when creating checkout sessions

Reviewer: It feels like both of these items should be in "Internal Changes and Upcoming Features", as checkout sessions are not going to be generally available in this release.

* Update - Remove the deferred intent flag (`wc-stripe-is-deferred-intent`) and the non-deferred intent code path
* Update - Deprecate unused non-deferred intent methods in WC_Stripe_UPE_Payment_Gateway

> * Update - Remove all deprecated code up to version 10.1.0

Reviewer: This almost certainly needs to be the first item in the list, given that there may be pretty major compatibility issues that come from the code removals.

> * Update - Update Stripe API to 2025-09-3.clover and dynamic Javascript to Clover

Reviewer: There is a typo in this item (and probably in `changelog.txt` as well.
```suggestion
* Update - Update Stripe API to 2025-09-30.clover and dynamic Javascript to Clover
```

> * Update - Reconfigure webhooks to include Checkout Session events when Adaptive Pricing is enabled via settings

Reviewer: This feels like another internal change/upcoming feature.

> * Update - Add iDEAL | Wero co-branding as Stripe iDEAL-to-Wero migration Phase 1

Reviewer: It feels like this entry could be clearer.
```suggestion
* Update - Use iDEAL | Wero co-branding to comply with branding changes
```

**Other Fixes and Updates**
* Fix - Update deprecated WooCommerce block checkout hook from `onCheckoutAfterProcessingWithSuccess` to `onCheckoutSuccess` in the saved token handler
* Fix - Fix currency formatting issue with ISK for express checkouts

> * Fix - Only load Stripe JS (not the full UPE bundle) on product and cart pages when express checkout is disabled

Reviewer: I think this qualifies as an important update given the performance impact.

> * Fix - Fix script dependencies and script cache versioning for classic checkout

Reviewer: I think this also qualifies as an important update because it can impact what scripts are being enqueued in classic checkout.

* Fix - Remove transient style caching in favor of a simple in-memory frontend cache
* Fix - Improve Stripe element appearance on non-checkout pages
* Tweak - Update PHPDoc in admin REST controllers and related code
* Tweak - Improve PHPDoc for migration and notes; minor notes refactor
* Tweak - Update PHPDoc for express checkout classes, block support class, and intent controller
* Tweak - Update PHPDoc for UPE payment method classes
* Tweak - Refactor order helper class to use constants for lock meta keys and reduce code duplication in meta getters
* Tweak - Update PHPDoc and fix minor issues for subscriptions and pre-order compatibility
* Tweak - Add missing JSDoc comments to client-side utility, API, and payment-flow functions

**Internal Changes and Upcoming Features**
* Dev - Remove unit tests that no longer make sense after recent file removals
* Dev - Deprecate old unused feature flag backend methods and remove related code from the frontend
* Dev - Remove unused frontend code related to block checkout

> * Dev - Remove unused frontend code related to UPE

Reviewer: This may need to be moved to Other Fixes and Updates, as there are possible implications for developer expectations.

* Dev - Fix WC beta version resolution in tests

> * Dev - Upgrade @stripe/react-stripe-js to ^5.4.1 and @stripe/stripe-js to ^8.6.0 in JavaScript dependencies

Reviewer: This also feels like it should move to "Other Fixes and Updates", as it may have impacts on rendering and styles.

* Dev - Fixes becs e2e tests
* Dev - Upgrade @types/react to ^18.3.7 in JavaScript dependencies
* Dev - Add CodeRabbit configuration with Stripe-focused review guidance
* Dev - Expand AI agent guidance with directory-level AGENTS and CLAUDE context files
* Dev - Update dependencies for building translations
* Dev - Extract shared banner styled components and dismissNotice utility to eliminate frontend duplication
* Dev - Use PaymentGatewayFeature constants instead of raw strings for payment gateway feature declarations in UPE payment method classes and traits

> * Update - Move class instantiations from their definition files to the plugin initialization code

Reviewer: This should also move to "Other Fixes and Updates", as it could have compatibility impacts on weird code paths.

* Add - Display adaptive pricing currency selector on blocks checkout page
* Add - Display adaptive pricing currency selector on classic checkout page
* Add - Add method to check if adaptive pricing is supported and pass the flag to checkout page
* Add - Add the base CSV feed for agentic commerce

> * Add - Backend support for processing payments with Checkout Session when using Adaptive Pricing.

Reviewer: Nit: This changelog should not have a trailing `.`.
```suggestion
* Add - Backend support for processing payments with Checkout Session when using Adaptive Pricing
```

== Changelog ==

= 10.5.0 - xxxx-xx-xx =

**Important Fixes and Updates**
* Update - Remove all deprecated code up to version 10.1.0
* Update - Update Stripe API to 2025-09-30.clover and dynamic Javascript to Clover
* Fix - Allow subscription edits when mandates are present, except for mandates from India
* Fix - Prevent fatal error when order ID in webhook references a refund
* Fix - Hide all payment methods on Add Payment Method page when saved payment methods are disabled
* Fix - Only load Stripe JS (not the full UPE bundle) on product and cart pages when express checkout is disabled
* Fix - Fix script dependencies and script cache versioning for classic checkout
* Update - Remove the deferred intent flag (`wc-stripe-is-deferred-intent`) and the non-deferred intent code path
* Update - Deprecate unused non-deferred intent methods in WC_Stripe_UPE_Payment_Gateway

> * Update - Use iDEAL | Wero co-branding to comply with branding changes

Reviewer: Nit: this entry has diverged from the entry in `changelog.txt`. We should make sure they are the same.

* Fix - Fix currency formatting issue with ISK for express checkouts
* Fix - Remove transient style caching in favor of a simple in-memory frontend cache
* Fix - Improve Stripe element appearance on non-checkout pages
* Fix - Remove WooCommerce session creation on product page load to improve cacheability
* Tweak - Update PHPDoc in admin REST controllers and related code
* Tweak - Improve PHPDoc for migration and notes; minor notes refactor
* Tweak - Update PHPDoc for express checkout classes, block support class, and intent controller
* Tweak - Update PHPDoc for UPE payment method classes
* Tweak - Refactor order helper class to use constants for lock meta keys and reduce code duplication in meta getters
* Tweak - Update PHPDoc and fix minor issues for subscriptions and pre-order compatibility
* Tweak - Add missing JSDoc comments to client-side utility, API, and payment-flow functions

**Internal Changes and Upcoming Features**
* Dev - Remove unit tests that no longer make sense after recent file removals
* Dev - Deprecate old unused feature flag backend methods and remove related code from the frontend
* Dev - Fix WC beta version resolution in tests

> * Dev - Fixes becs e2e tests

Reviewer: Not sure how this happened, but `changelog.txt` has this as `* Dev - Fix becs e2e tests`. We should ensure they match.
```suggestion
* Dev - Fix becs e2e tests
```
